### ISSUE TITLE
Sort requests in folder, alphabetically

### ISSUE THREAD

**Issue by karlmacklin** [a85/POSTMan-Chrome-Extension#5222]
I would like to have a button to sort the requests in a collection folder, **alphabetically**, with a single click. 

The current sorting is manual and if you have a large collection of requests that you have created over time, that big mess could quickly be solved by such a button, and not so quickly by manual work dragging and dropping.

**Comment by hanikhan** [comment]
This would definitely be a great feature to have

**Comment by DeadDuck** [comment]
I see this sort button in the docs which doesn't appear on my setup. Was it removed or is it just a concept?

![image](https://user-images.githubusercontent.com/474456/52614150-d33b4500-2ea1-11e9-9881-330de37afc2f.png)

**Comment by qwertycody** [comment]
@DeadDuck, you're not alone. I am having same issue. +1

**Comment by johnli2210** [comment]
@DeadDuck Yep, I'm also seeing this issue. It looks like this feature is available in the Chrome App, but not on native Desktop App (for Mac OS at least) yet. I guess it just hasn't yet been ported over; perhaps it's harder to implement on Desktop?


![Postman_and_Postman](https://user-images.githubusercontent.com/26591164/57427197-43215780-7266-11e9-8f99-c4b633645b5e.png)

**Comment by glau2** [comment]
+1 on adding ability to sort in both collection level and folder level

**Comment by badmansan** [comment]
Any plans to add this feature?

**Comment by yashbhutwala** [comment]
Any updates on this, much desired feature, thanks!

**Comment by Alexand00r** [comment]
Also looking forward this feature in desktop app. Thank you in advance

**Comment by penguinawesome** [comment]
+1!

**Comment by GXelha** [comment]
+1!

**Comment by mattiagalati** [comment]
+1

**Comment by josh-kimmm** [comment]
+1

**Comment by chris-rl** [comment]
+1

**Comment by JSmythSSG** [comment]
+1

**Comment by Paridhi-Mohindra** [comment]
+1

**Comment by alexxozo** [comment]
+1

**Comment by danielrobleM** [comment]
+1

**Comment by minhthong095** [comment]
+1

**Comment by randy-ja** [comment]
+1

**Comment by solsticesurfer** [comment]
+1. I have collections with 100s of entries. Manually sorting through drag-and-drop is difficult and time-consuming. Hell, make it a premium feature if you must but sorting is sorely needed.

**Comment by gusandrioli** [comment]
Also, It'd be great to have a sort for JSON data inside each request too.

**Comment by donreinardgomez** [comment]
+1

**Comment by annaelizabeth2019** [comment]
+1

**Comment by daniele-orlando** [comment]
### PLEASE, STOP POSTING

> +1

If you want to express your agreement, use the thumb 👍 on the **initial post**.

![Schermata 2021-04-01 alle 12 40 32](https://user-images.githubusercontent.com/504386/113292462-d6e5ec00-92f4-11eb-9b81-ec0d8ba8fca6.png)

**Comment by matthewlein** [comment]
I also want this, so I knocked out a quick npm module to sort an exported collection by name and verb.

https://github.com/matthewlein/postman-collection-sorter

## Usage

1. In Postman, export your collection as a Collection v2.1
1. Run:

   `npx @matthewlein/postman-collection-sorter ./path/to/your/API.postman_collection.json`

1. You should see output like:

```
Sorting collection file:  ./path/to/your/API.postman_collection.json
Sorting Successful!
Sorted collection saved to: ./path/to/your/API.postman_collection.sorted.json
```

4. In Postman, import the new sorted JSON file.
1. Enjoy that sweet sweet alphabetical sorting.

**Comment by clarkwang** [comment]
> I also want this, so I knocked out a quick npm module to sort an exported collection by name and verb.
> 
> https://github.com/matthewlein/postman-collection-sorter

Inspired by this I wrote a simple Python script at https://github.com/clarkwang/postman-sort .

**Comment by JohnCarmichael3000** [comment]
> I also want this, so I knocked out a quick npm module to sort an exported collection by name and verb.
> 
> https://github.com/matthewlein/postman-collection-sorter
> 
> ## Usage
> 1. In Postman, export your collection as a Collection v2.1
> 2. Run:
>    `npx @matthewlein/postman-collection-sorter ./path/to/your/API.postman_collection.json`
> 3. You should see output like:
> 
> ```
> Sorting collection file:  ./path/to/your/API.postman_collection.json
> Sorting Successful!
> Sorted collection saved to: ./path/to/your/API.postman_collection.sorted.json
> ```
> 
> 1. In Postman, import the new sorted JSON file.
> 2. Enjoy that sweet sweet alphabetical sorting.

Thanks you for this solution, worked like a charm. Melts my mind that this is required and not a button yet, though.

**Comment by dh-pt** [comment]
...how is it possible that you can't yet sort collections in almost 2022?

**Comment by rishinaveen** [comment]
> I also want this, so I knocked out a quick npm module to sort an exported collection by name and verb.
> 
> https://github.com/matthewlein/postman-collection-sorter
> 
> ## Usage
> 1. In Postman, export your collection as a Collection v2.1
> 2. Run:
>    `npx @matthewlein/postman-collection-sorter ./path/to/your/API.postman_collection.json`
> 3. You should see output like:
> 
> ```
> Sorting collection file:  ./path/to/your/API.postman_collection.json
> Sorting Successful!
> Sorted collection saved to: ./path/to/your/API.postman_collection.sorted.json
> ```
> 
> 4. In Postman, import the new sorted JSON file.
> 5. Enjoy that sweet sweet alphabetical sorting.

Great little utility. thanks a lot for sharing the tool and appreciate the time you spent creating it.

**Comment by clarkwang** [comment]
I wrote a simple Python script to sort the exported JSON file and then the sorted JSON can be imported back to Postman. Those who are interested in can give it a try. The script is at https://github.com/clarkwang/postman-sort .

**Comment by svenbeck27** [comment]
+1

**Comment by innotech-manar-ishtayeh** [comment]
+1

**Comment by JohnCarmichael3000** [comment]
There have been dozens and dozens of releases since this issue was opened. So at this point one must assume that adding this feature isn't in the cards for them?

**Comment by dh-pt** [comment]
> There have been dozens and dozens of releases since this issue was opened. So at this point one must assume that adding this feature isn't in the cards for them?

Feels pretty standard and expected, though... like a very basic feature missing.

**Comment by aidar87** [comment]
Guys, we need this sorting!!!

**Comment by wnmurphy** [comment]
+1 for sorting collections/requests alphabetically in desktop client pls.

**Comment by DeeplySheeply** [comment]
+1

**Comment by KGSands** [comment]
+1

**Comment by tsong-wallis** [comment]
> ...how is it possible that you can't yet sort collections in almost 2022?

we are way into 2022 and still missing sort feature

**Comment by karlmacklin** [comment]
Good news everyone!
Since the creation of this issue I have now managed to sort all my requests manually, so currently things are in order.

Nevertheless, I would have saved many years of work if this feature was implemented, and it can probably save some time for other people as well. So I still vouch for the implementation of it!

**Comment by Justintime50** [comment]
Postman never ceases to amaze me for its lack of basic features like auto-save and sorting folders... 

This should definitely be something that gets added. It's mind boggling that four years later it still hasn't been implemented.

**Comment by madalina-gramaa** [comment]
pretty pleaaaase, let's have this 🙏

**Comment by Pedroprt** [comment]
Postman is a great tool, but that's unbeliveale that this basic feature doesn't exit yet.
Please add a requests inside collection sort option.

**Comment by absieee** [comment]
I asked Santa to get me this feature for Christmas this year...let's see what happens.

**Comment by naufalahmadfauz** [comment]
I just can't believe that i had to search for this thing 😂

**Comment by solsticesurfer** [comment]
Sorry, Timmy. There is no Santa Claus. :/On Dec 8, 2022, at 7:24 AM, Abhishek Sharma ***@***.***> wrote:﻿
I asked Santa to get me this feature for Christmas this year...let's see what happens.

**Comment by karlmacklin** [comment]
New status update:

First, the year was 2018. I encountered the issue of having a bunch of unsorted requests in my folders, so I created this issue.
Due to the lack of development on this issue I started the work of manually sorting them.

Then, in August of 2022 I had finally succeeded. Calm ensued and there was peace once again.

But then. Summer vacation ended. New clients appeared, slowly creeping towards me almost zombie-like chanting "requests... requests..." - they all needed new endpoints tested and parameters verified.

I regret to inform that the status has now changed to the previous state:
**My requests are once again unsorted.**

If I just had an option to automatically sort these requests alphabetically, the world would be a better place.

**Comment by giridharvc7** [comment]
@karlmacklin, we've introduced a way you can do bulk manual reordering. Have you tried this feature? 
For sorting itself, we are looking into this, and I want to understand if you want your workflow a bit better.

**Comment by karlmacklin** [comment]
> @karlmacklin, we've introduced a way you can do bulk manual reordering. Have you tried this feature? For sorting itself, we are looking into this, and I want to understand if you want your workflow a bit better.

Hi, not sure, is "bulk manual reordering" the same as dragging and dropping items within a collection? Or something else? I checked the context menu and did not see anything specific there.

**Comment by giridharvc7** [comment]
Yeah, its the same as multi-selecting a bunch of items, dragging and dropping. 
https://user-images.githubusercontent.com/1040583/216223083-f1386a55-3f7a-4578-8f26-931756afeedf.mov

**Comment by solsticesurfer** [comment]
It’s only been five years. You have to give these things time. Sorting is harrrrrrd. 😀On Jan 23, 2023, at 2:40 PM, karlmacklin ***@***.***> wrote:﻿
New status update:
First, the year was 2018. I encountered the issue of having a bunch of unsorted requests in my folders, so I created this issue.
Due to the lack of development on this issue I started the work of manually sorting them.
Then, in August of 2022 I had finally succeeded. Calm ensued and there was peace once again.
But then. Summer vacation ended. New clients appeared, slowly creeping towards me almost zombie-like chanting "requests... requests..." - they all needed new endpoints tested and parameters verified.
I regret to inform that the status has now changed to the previous state:
My requests are once again unsorted.
If I just had an option to automatically sort these requests alphabetically, the world would be a better place.

**Comment by clarkwang** [comment]
I wrote a simple Python script to sort the exported JSON file and then the sorted JSON can be imported back to Postman. Those who are interested can give it a try. The script is at https://github.com/clarkwang/postman-sort .

Note that it sorts not only requests, but also collection vars, request's params, folders and examples.

**Comment by ruslaniv** [comment]
Just checking in at year 2023, 5 years later and still no sorting. This is quite astounding! Wow! 
I'm definitely NOT upgrading free plan now although I was considering it.

**Comment by karlmacklin** [comment]
> Yeah, its the same as multi-selecting a bunch of items, dragging and dropping. https://user-images.githubusercontent.com/1040583/216223083-f1386a55-3f7a-4578-8f26-931756afeedf.mov

Sorry for the slow reply.

This isn't really close to the reason for this request in the first place.

Look, it's kind if simple really - some users have multiple (maybe 20+) requests in a single collection. They might work on it for a bit, and then **come back later** at which point they don't have immediate memory of where which request is in the list.

How do we humans usually scan lists of text really fast? By browsing it alphabetically.

There are two reasons then for this request:

1. We want to have our request lists within collections alphabetically sorted.
2. We do not want manually (multi-selection or not) spend time sorting requests, as this is something that is expected as a common feature in software.

**Comment by DeadDuck** [comment]
@karlmacklin  I have 140+ in one collection, and 200-300 in another, spanned into many folders.

5 years for a simple sort operation, which would take a few hours at most. Does anybody from the team really see this issue, or are we writing into a black hole? Anybody from Postman? ;-)

**Comment by DeadDuck** [comment]
>  For sorting itself, we are looking into this, and I want to understand if you want your workflow a bit better.

Oh, just noticed someone from Postman actually sent a response! Sorry. I have a full-time job, but here is my suggestion: Hire me part-time. Like, buy me dinner and I'll sort out this long-time issue for you, @giridharvc7. 25 years of experience in programming for the desktop and the web. Fair offer.

**Comment by fndribeiro** [comment]
And here I am sorting my requests manually. @_@

**Comment by mikeroque** [comment]
Postman devs, are you planning to implement this feature or not?

Respectfully, this is a pretty easy feature to add that many users want, so I think it's reasonable for you to either do it or tell us why you won't.

Thanks!

**Comment by MariuzM** [comment]
Jezus cryst postman dev's what are you doing over there, is this a dead app should we stop using it?

**Comment by VanPuyveldeKevin** [comment]
At this point it's Postman DEVs literally saying screw you to their community. After all, having a manual drag/drop 'move' feature and utterly refusing an auto-sort feature just boggles the mind. The auto-sort literally just needs to make use of existing application features, but then do it on a loop of some kind (it's basically the move operation, but in bulk).

It's honestly one of the most basic of features for any app that has any kind of UI. Just imagine your OS forcing you to manually sort each and EVERY single folder and file on your hard drive. You'd have a fulltime job just maintaining the sort order.

We're not even asking a lot here, just asking for alphabetical order in the collection tree.

**Comment by MariuzM** [comment]
Yeah i moved to Rest Client for vscode

**Comment by DeadDuck** [comment]
Yeah. I'm an extremely patient person in general. But this sounds like a single-man operation, the single man being exhausted for years. I even offered support. 🤷‍♂️

**Comment by misternian** [comment]
Five years later, they still haven't finished this simple feature

**Comment by nguyenngocqui-official** [comment]
any updates on this feature?

**Comment by QADovah** [comment]
@giridharvc7 , can you update us the status of this ticket? 

As @karlmacklin, I add my example.

-Already working with MANUALLY sorted folders and subfolders.
-New wild endpoint appears on swagger.
-Need to test the endpoint.
-Create new folder below the first letter of the folder I want.
-The new folder creates after the LAST folder. (Guess that is a new issue).
-Need to CTRL + A to select the full name to be able to rename, is not like windows (creating the folder with no string inside (New issue here?)
-Type que folder name. And then the sub folders to add Folders to CRUD every endpoint. 
-NEED TO MOVE THE FOLDER MANUALLY TO SORT.

And I`m just the only tester. If a dev starts to creating cases, or more testers are assigned to this project it would be A LOT.

**Comment by Alan0x01** [comment]
Is there any progress on this requirement? In the planning or development?

**Comment by MariuzM** [comment]
I personally just stopped using Postman, guys try Yaak.app i find it much better

**Comment by ultrabstrong** [comment]
This would be a very helpful feature and it is very easy to implement. The community has been asking for this for a really, really, really long time! I would also like to hear an update on this and see progress on adding it.

**Comment by DeadDuck** [comment]
Amazing!

**Comment by DeadDuck** [comment]
I even offered free professional support. The issue is not even about development resources...

**Comment by jCrip** [comment]
ALso would be great to sort by request method

**Comment by fndribeiro** [comment]
I basically gave up on Postman and started using Insomnia.

**Comment by jnovak-SM2Dev** [comment]
Are the devs ever going to add this? It's crazy something so simple still isn't properly added.

**Comment by karlmacklin** [comment]
> Are the devs ever going to add this? It's crazy something so simple still isn't properly added.

Nah it has become like an inside joke/meme now basically, it would be weird to actually implement it.

This is the issue that shall live on! My magnum opus!

**Comment by dongockhanh3103** [comment]
move to insomna this time

**Comment by scastria** [comment]
Not quite the solution everybody is looking for, but you CAN do it with the Postman Admin REST API and my postman Terraform provider: https://registry.terraform.io/providers/scastria/postman/latest/docs/resources/collection_sort. My postman TF provider is nothing more than a TF wrapper around the Postman Admin REST API.  So to do it with the API, you would get your collection which returns the entire JSON description of your collection and all of its contents.  You could run it through a JSON sorter and then run a PUT to write it back to Postman.

If you are a fan of Terraform, you might want to give it a look.  I use it for all of my companies Postman automation.  I have TF reading the openapi spec and creating collection, folders, requests (all sorted) based on that spec.  When my spec changes, the TF updates the Postman collection to match.

**Comment by sidvermapostman** [comment]
Thanks everyone for sharing your thoughts on this! **We hear you** and we're looking at sidebar improvements as whole. This will be released soon. We’ll share updates as soon as we have more to share.

**Comment by absieee** [comment]
Finally 7 years later this is getting actually picked up? For those who've been subscribed and invested for so long, guys, we need to celebrate.

**Comment by JimTapp** [comment]
> Thanks everyone for sharing your thoughts on this! **We hear you** and we're looking at sidebar improvements as whole. This will be released soon. We’ll share updates as soon as we have more to share.

This seems like great news, but was this comment referencing Agent Mode?  If so, I tested this, and Agent Mode can't alphabetize folder contents.  Sometimes, it will tell you that it can and even that it did, when you ask it too (but it does not). Other times it correctly states that it can't alphabetize.  Either way, if Agent Mode was meant to be the solution for this, at the moment, it isn't.

I still imagine being able to right click on a folder or collection or click the 3 dot menu on them, and choosing "Alphabetize all items".  Or even an option in the general settings of the app, to alphabetize by default!  At least for our use case, we would always want everything alphabetical anyways.

**Comment by sidvermapostman** [comment]
@JimTapp We will be exposing Agent Mode to be able to sort collection and folder within collections however, to solve for this issue we are adding an entry in the triple dot menu for collections/folders accordingly.

**Comment by sidvermapostman** [comment]
Hey folks, we've released the fix to this. Take a look at the attach video for details. I'll close out this issue now, thank you all for raising!

https://github.com/user-attachments/assets/e449e668-490c-4fdc-8b21-b2d802b71e84

**Comment by ruslaniv** [comment]
Welp, it only took 7 years! Like my grandma used to say 

> patience is bitter but its fruit is sweet

Although I doubt I taste the sweet since we abandoned Postman long time ago ))
_partially because of this exact issue_

**Comment by karlmacklin** [comment]
If anyone has any tips on how to celebrate this please let me know as my soul is now hollow.

**Comment by JimTapp** [comment]
Hi, @sidvermapostman - while this is a step in the right direction, it seems that it's only changing the _view_, and it's only for _you_.  We need to permanently update the sort order; and have it affect _everyones_ view; otherwise this isn't really a fix for us and still requires manual interaction from each individual user to get it to function like, well, how it should by default tbh.  

How can we make this alphabetical sort order permanent and not a temporary view?
